src/controllers/krate/search.rs

```diff
@@ -551,29 +557,30 @@ mod seek {
             New(#[serde(with="ts_microseconds")] chrono::NaiveDateTime, i32)
             RecentUpdates(#[serde(with="ts_microseconds")] chrono::NaiveDateTime, i32)
             RecentDownloads(Option<i64>, i32)
-            Downloads(i32, i32)
+            Downloads(i64, i32)
             Query(bool, i32)
             Relevance(bool, f32, i32)
         }
     }
 
     impl Seek {
-        pub(crate) fn to_payload(&self, record: &(Crate, bool, Option<i64>, f32)) -> SeekPayload {
+        pub(crate) fn to_payload(
+            &self,
+            record: &(Crate, bool, i64, Option<i64>, f32),
```

Comment:
From a pure maintainability perspective I don't really love the 5-tuple. It was sort of almost OK when all the fields had significantly distinct types, but now there's an `i64` and an `Option<i64>` it's starting to feel a little muddy.

I'm not sure this _needs_ to change immediately — presumably we'd have to introduce a new `struct` that implements `From<(Crate, bool, i64, Option<i64>, f32)>` or similar, and that might not be a pure win — but I'm worried that someone (OK, me) is eventually going to forget which numbered field is which.

Reply:
yeah, I agree about that. I mentioned something similar in https://github.com/rust-lang/crates.io/pull/8037#discussion_r1499030402.

@eth3lbert is this something you could look into? :)

Reply:
Indeed, I think I can modify the macro to add support for braced structs. Then, I would like to know if we want to continue supporting tuple structs.

Reply:
> I would like to know if we want to continue supporting tuple structs.

I think it would be fine to move everything over to structs with named fields and remove support for tuples